packages/angular/cli/src/utilities/environment-options.ts

```diff
@@ -6,28 +6,63 @@
  * found in the LICENSE file at https://angular.dev/license
  */
 
-function isPresent(variable: string | undefined): variable is string {
-  return typeof variable === 'string' && variable !== '';
-}
+/** A set of strings that are considered "truthy" when parsing environment variables. */
+const TRUTHY_VALUES = new Set(['1', 'true']);
 
-function isDisabled(variable: string | undefined): boolean {
-  return isPresent(variable) && (variable === '0' || variable.toLowerCase() === 'false');
-}
+/** A set of strings that are considered "falsy" when parsing environment variables. */
+const FALSY_VALUES = new Set(['0', 'false']);
 
-function isEnabled(variable: string | undefined): boolean {
-  return isPresent(variable) && (variable === '1' || variable.toLowerCase() === 'true');
+/**
+ * Checks if an environment variable is present and has a non-empty value.
+ * @param variable The environment variable to check.
+ * @returns `true` if the variable is a non-empty string.
+ */
+function isPresent(variable: string | undefined): variable is string {
+  return typeof variable === 'string' && variable !== '';
 }
 
-function optional(variable: string | undefined): boolean | undefined {
+/**
+ * Parses an environment variable into a boolean or undefined.
+ * @returns `true` if the variable is truthy ('1', 'true').
+ * @returns `false` if the variable is falsy ('0', 'false').
+ * @returns `undefined` if the variable is not present or has an unknown value.
+ */
+function parseTristate(variable: string | undefined): boolean | undefined {
   if (!isPresent(variable)) {
     return undefined;
   }
 
-  return isEnabled(variable);
+  const value = variable.toLowerCase();
+  if (TRUTHY_VALUES.has(value)) {
+    return true;
+  }
+  if (FALSY_VALUES.has(value)) {
+    return false;
+  }
+
+  return undefined;
 }
```

Comment:
Consider: Could we remove the "tristate" concept by just providing a default value like `parseEnvBoolean(process.env['FOO'], true /* default */)`?